### PR TITLE
Bump matrix tests to oldest and newest Python versions currently supported in CI

### PR DESCRIPTION
**Pull request by joanise**

### CODE REVIEW

File: .github/workflows/test-suite.yml [hunk 1]
```diff
@@ -8,15 +8,15 @@ jobs:
     strategy:
       max-parallel: 5
       matrix:
-        python: [3.6, 3.9]
+        python: ["3.7", "3.12"]
```
Comment (SamuelLarkin) [inline .github/workflows/test-suite.yml:11]: Don't we want to test 3.7, 3.8, 3.9, 3.10, 3.11 & 3.12?

**Comment by joanise** [comment]
PortageClusterUtils will also need its actions versions bumped, like in [3f89b04](https://github.com/nrc-cnrc/PortageTextProcessing/pull/5/commits/3f89b043493d6f59b6a9b9d445f433000a7c75d1)

**Review by SamuelLarkin — APPROVED** [review]
Quite simple.

File: .github/workflows/test-suite.yml (hunk 1 above)
Comment (joanise) [reply .github/workflows/test-suite.yml:11]: We could, but it didn't seem necessary. It's not like we have complex Python stuff in there, it's mostly Perl and others anyway. It didn't seem worth the cycles.

Comment (SamuelLarkin) [reply .github/workflows/test-suite.yml:11]: For the environment 👍🏼

**Pull request merged**